crates/synthesis/src/linalg/cos_sin_decomp.rs

```diff
@@ -120,12 +75,10 @@ pub fn cos_sin_decomposition(u: DMatrix<Complex64>) -> CosSinDecompReturn {
     // We will first find l0, c, and r0, then s and l1, and finally r1.
 
     // Apply SVD to u00
-    let svd = u00
-        .try_svd(true, true, 1e-12, 0)
-        .expect("Problem with SVD decomposition");
-    let mut l0 = svd.u.unwrap();
-    let mut r0 = svd.v_t.unwrap();
-    let mut c: DVector<f64> = svd.singular_values.column(0).into_owned();
+    let svd = u00.svd().expect("Problem with SVD decomposition");
+    let l0 = svd.U().to_owned();
+    let r0 = svd.V().adjoint().to_owned();
```

Comment:
You immmediately reverse the rows or cols and make another copy here. You can save the allocation here and only copy once when reversing the rows or cols.

```suggestion
    let l0 = svd.U();
    let r0 = svd.V().adjoint();
```

Reply:
Done in [501807e](https://github.com/Qiskit/qiskit/pull/14967/commits/501807e44530dd6f0ea2ab2c2baa92fba924895f)